components/src/preact/components/error-display.tsx

```diff
@@ -1,5 +1,20 @@
 import { type FunctionComponent } from 'preact';
 
+export class UserFacingError extends Error {
+    constructor(message: string) {
+        super(message);
+        this.name = 'UserFacingError';
+    }
+}
+
 export const ErrorDisplay: FunctionComponent<{ error: Error }> = ({ error }) => {
-    return <div>Error: {error.message}</div>;
+    const errorTypesToShow = ['UserFacingError'];
+
+    return (
+        <div className='h-full w-full rounded-md border-2 border-gray-100 p-2 flex items-center justify-center flex-col'>
+            <div className='text-red-700 font-bold'>Error</div>
+            <div>Oops! Something went wrong.</div>
+            {errorTypesToShow.includes(error.name) && <div className='text-sm text-gray-600'>{error.message}</div>}
```

Comment:
```suggestion
            {error instanceof UserFacingError && <div className='text-sm text-gray-600'>{error.message}</div>}
```
Why don't we simply do this?

Reply:
True, but I wanted this to be extensible, so we may define many different UserFacingError types in the future.

Reply:
I don't know whether that is the right way. If you want to extend it, you can still create a new child class of `UserFacingError`.